importer/api/__init__.py

```diff
@@ -136,3 +147,24 @@ def import_children(course, element, parent):
     for child in element.getchildren():
         if child.tag in DESCRIPTOR_TAGS:
             import_children(course, child, resource)
+
+
+def import_static_assets(path, course):
+    """
+    Upload all assets and create model records of them for a given
+    course and path.
+
+    Args:
+        path (string_type): course specific path to extracted OLX tree.
+        course (learningresources.Course): Course to add assets to.
+
+    Returns:
+        None
+    """
+    for root, _, files in walk(path):
+        for name in files:
+            with open(join(root, name), 'r') as open_file:
+                django_file = File(open_file)
+                # Remove base path from file name
+                django_file.name = join(root, name).replace(path + sep, '')
```

Comment:
For what it's worth, `splitext` may help here. It looks like it puts the pathsep on the "wrong side" for what you want, but it's a bit more explicit than `replace`.

```python
from os.path import splitext
filename = "something.txt"
result = splitext(filename)
print result
```


Output: `('something', '.txt')`